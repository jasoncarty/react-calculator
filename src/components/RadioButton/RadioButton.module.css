.radioButton {
  display: flex;
  flex-direction: row;
  justify-content: left;
  align-items: center;
  height: 25px;
}

.text {
  margin-left: 15px;
}
