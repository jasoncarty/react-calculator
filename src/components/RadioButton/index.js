import React from "react";
import PropTypes from "prop-types";

import styles from "./RadioButton.module.css";

const RadioButton = ({ text, onChange, value, ...rest }) => (
  <div className={styles.radioButton}>
    <input
      type="radio"
      name="calculationType"
      value={value}
      onChange={onChange}
      {...rest}
    />
    <div className={styles.text}>{text}</div>
  </div>
);

RadioButton.propTypes = {
  text: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default RadioButton;
