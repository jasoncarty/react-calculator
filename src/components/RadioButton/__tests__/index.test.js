import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import RadioButton from "../index";

describe("<RadioButton />", () => {
  it("matches snapshot", () => {
    const instance = ReactTestRenderer.create(
      <RadioButton
        text="lkjasdfl"
        onChange={() => {}}
        value="mockValue"
        text="lkajsdflkjasfd"
        someOtherProp={() => {}}
      />
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
