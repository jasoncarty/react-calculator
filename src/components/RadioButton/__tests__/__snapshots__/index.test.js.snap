// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> matches snapshot 1`] = `
<div
  className="radioButton"
>
  <input
    name="calculationType"
    onChange={[Function]}
    someOtherProp={[Function]}
    type="radio"
    value="mockValue"
  />
  <div
    className="text"
  >
    lkajsdflkjasfd
  </div>
</div>
`;
